facebooknuclide/nuclide-format-js
# Update to latest jscodeshift

By updating to latest jscodeshift:

1) This fixes the iife test error that was caused by a bug in recast, fixed in: https://github.com/benjamn/recast/pull/410
2) This fixes support for bare function arguments, which were breaking recast due to a bug in babylon, fixed in: https://github.com/babel/babylon/pull/565 and adds a test for the corresponding syntax

I have also removed dead dependencies from `package.json`. Note why the remaining dep is not locked down, explained to me by @bestander:

> using 1.2.3 instead of ^1.2.3 does not give you stability actually, in many cases it is actually worse 🙂
The problem is that ^ is the default for npm and all the packages that you depend on use ^.
So left-pad@1.2.3 will download the same left-pad but its dependencies are specified as the author of left-pad did, with ^.
So you'll have variance in subdependencies between runs.
That is why yarn.lock is needed.
**BTW, don't worry about npm, they use their own lock file by default too, now.**

Before we release all the updates, I want do two last things (ideally tomorrow):

1) Make sure the sorting is correct for both types and modules with our order-requires rule
2) Add a notification when the formatter fails due to a syntax error in user's source code, instead of silently failing

Plus:
- Compare perf with Flow parser
- Document use in codemods
